osbuild/util/containers.py

```diff
@@ -127,4 +148,32 @@ def container_source(image):
             os.symlink(image_filepath, tmp_source)
 
         image_source = f"{container_format}:{tmp_source}"
-        yield image_name, image_source
+        yield image_source
+
+
+@contextmanager
+def container_source(image):
+    image_filepath = image["filepath"]
+    container_format = image["data"]["format"]
+    image_name = image["data"]["name"]
+
+    if container_format not in ("dir", "oci-archive", "containers-storage"):
+        raise RuntimeError(f"Unknown container format {container_format}")
+
+    if container_format == "containers-storage":
+        with containers_storage_source(image, image_name, image_filepath, container_format) as image_source:
+            yield image_name, image_source
+    else:
+        with dir_oci_archive_source(image, image_name, image_filepath, container_format) as image_source:
+            yield image_name, image_source
+
+
+def get_host_storage(storage_conf):
```

Comment:
as mentioned about we could change this to `storage_conf=None` which feels slightly nicer